### PR TITLE
zq/reimpl log_softmax

### PR DESCRIPTION
**Pull request by NeosZhang**
## Motivation and Context

reimpl log_softmax

## Description

reimpl log_softmax

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Review by POI-WX — APPROVED** [review]

File: impl/ascend_npu/diopi_impl/logsoftmax.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include "helper.hpp"
+#include "op_plugin/AclOpsInterface.h"
+#include "op_plugin/OpApiInterface.h"
+#include "op_plugin/utils/op_api_common.h"
+namespace OP_IMPL_NS {
+diopiError_t diopiLogSoftmax(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input, int64_t dim) {
+    BEGIN_CALL_ACL_OP(input, out);
+    if (inputAt.numel() == 1) {
+        op_api::fill_(outAt, 0);
+        return diopiSuccess;
```
Comment (yangbofun) [inline impl/ascend_npu/diopi_impl/logsoftmax.cpp:16]: 这里看下为啥要调用fill。

**Review by hellozmz — APPROVED** [review]

File: impl/ascend_npu/diopi_impl/logsoftmax.cpp (hunk 1 above)
Comment (NeosZhang) [reply impl/ascend_npu/diopi_impl/logsoftmax.cpp:16]: 当输入为scalar且dim不为0时会报错，但pytorch结果为1

File: impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp [hunk 2]
```diff
@@ -1519,6 +1519,19 @@ at::Tensor OpPreparation::apply_tensor_with_format(c10::IntArrayRef sizes, const
         sizes, optTypeMetaToScalarType(options.dtype_opt()), options.layout_opt(), options.device_opt(), options.pinned_memory_opt(), fixFormat, keep_format);
 }
 
+inline at::Tensor apply_tensor_use_empty(c10::IntArrayRef sizes, const c10::TensorOptions& options) {
+    return at_npu::native::empty_npu(
+        sizes, options.dtype().toScalarType(), c10::nullopt, at::Device(c10::DeviceType::PrivateUse1), false, c10::MemoryFormat::Contiguous);
```
Comment (yangbofun) [inline impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp:1524]: 这里改为XLA

File: impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp [hunk 3]
```diff
@@ -1519,6 +1519,19 @@ at::Tensor OpPreparation::apply_tensor_with_format(c10::IntArrayRef sizes, const
         sizes, optTypeMetaToScalarType(options.dtype_opt()), options.layout_opt(), options.device_opt(), options.pinned_memory_opt(), fixFormat, keep_format);
 }
 
+inline at::Tensor apply_tensor_use_empty(c10::IntArrayRef sizes, const c10::TensorOptions& options) {
+    return at_npu::native::empty_npu(
```
Comment (zhaoguochun1995) [inline impl/ascend_npu/torch_npu/csrc/DIOPIAdapter.cpp:1523]: return at_npu::native::empty_npu(sizes, options);

**Review by yangbofun — APPROVED** [review]

**Pull request merged**